IntegrationTests/CredentialsProviders/ProcessCredentialsProviderIntegrationTests/ProcessCredentialsProviderTests.swift

```diff
@@ -0,0 +1,37 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import XCTest
+import AWSS3
+import AWSClientRuntime
+
+// Please provide your-access-key and your-secret-key in Resources/credenitals
+class ProcessCredentialProviderTests: XCTestCase {
+        
+    var client: S3Client!
+        
+    override func setUp() async throws {
+        // Setup ProcessCredentialsProvider
+        let processCredentialsProvider = try ProcessCredentialsProvider(
+            configFilePath: Bundle.module.path(forResource: "config", ofType: nil)!,
+            credentialsFilePath: Bundle.module.path(forResource: "credentials", ofType: nil)!
+        )
+
+        // Setup S3ClientConfiguration to use ProcessCredentialsProvider
+        let testConfig = try await S3Client.S3ClientConfiguration()
+        testConfig.credentialsProvider = processCredentialsProvider
+
+        // Initialize our S3 client with the specified configuration
+        client = S3Client(config: testConfig)
+    }
+
+    // This test calls listBuckets() and forces S3Client to use ProcessCredentialsProvider
+    func test_listBuckets() async throws {
+        _ = try await client.listBuckets(input: ListBucketsInput())
```

Comment:
Not sure how this test is going to pass, the credentials will be loaded from the `echo` command in the `credentials` config file below, which aren't real AWS credentials

Reply:
So we have a few options here. Tests aren't run automatically so we dont have a dedicated AWS Account that these are running against. So all the integration tests are running against the user's personal ~/.aws/config and ~/.aws/credentials from my understanding. User who runs this integ test likely does not have their credentials file setup to be using ProcessCredentialsProviders so either they would have to modify their own or use this one which I provided for convenience. Does that make sense?